services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java

```diff
@@ -0,0 +1,221 @@
+package org.keycloak.truststore;
+
+import org.jboss.logging.Logger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Builds a system-wide truststore from the given config options.
+ */
+public class TruststoreBuilder {
```

Comment:
I'm not sure about initialization order - are we guaranteed that the spi trustsore will be initialized before drivers or any other customization?

Reply:
Yeah, it might be a bit problematic as the `KeycloakSessionFactory` is initialized in `QuarkusKeycloakApplication` which might be way too late. If that's the case, let's keep as is.

Reply:
Yep, it'd be good to solve it in a follow-up PR.